components/datetime/src/provider/timezones.rs

```diff
@@ -0,0 +1,84 @@
+use std::borrow::Cow;
+
+pub type LiteMap<K, V> = std::collections::BTreeMap<K, V>;
+//use litemap::LiteMap;
+
+macro_rules! map_access {
+    ($outer: ty => $inner: ty) => {
+        impl $outer {
+            pub fn get<Q: ?Sized>(&self, key: &Q) -> Option<&$inner>
+            where
+                Q: Ord,
+                Cow<'static, str>: std::borrow::Borrow<Q>,
+            {
+                self.0.get(key)
+            }
+        }
+
+        impl<Q: ?Sized> std::ops::Index<&Q> for $outer
+        where
+            Q: Ord,
+            Cow<'static, str>: std::borrow::Borrow<Q>,
+        {
+            type Output = $inner;
+            fn index(&self, key: &Q) -> &Self::Output {
+                &self.0[key]
+            }
+        }
+    };
+}
+
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct TimeZoneFormatsV1 {
+    pub hour_format: Cow<'static, str>,
+    pub gmt_format: Cow<'static, str>,
+    pub gmt_zero_format: Cow<'static, str>,
+    pub region_format: Cow<'static, str>,
+    pub region_format_variants: LiteMap<Cow<'static, str>, Cow<'static, str>>,
+    pub fallback_format: Cow<'static, str>,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct TimeZoneNamesLongV1(pub LiteMap<Cow<'static, str>, Cow<'static, str>>);
+map_access!(TimeZoneNamesLongV1 => Cow<'static, str>);
+
+#[derive(PartialEq, Debug, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct TimeZoneNamesShortV1(pub LiteMap<Cow<'static, str>, Cow<'static, str>>);
```

Comment:
Love this. I was thinking they had to be 1:1 per key.

Reply:
I tried this, but I don't think I can make it work. The erased provider implementation needs to be able to determine the return type via turbofish:

```rust
key::TIMEZONE_GENERIC_NAMES_LONG_V1 => {
                let mut result: DataResponse<MetaZoneGenericNamesLongV1> =
...
}
```

So I don't think I can just have both of these be represented in a single `MetaZoneGenericNamesV1` struct, otherwise it would be ambiguous which payload to load based on the type.